executors/blsKeysExecutor.go

```diff
@@ -0,0 +1,162 @@
+		}
+	}
+	if check.IfNil(args.ValidatorStatisticsQuerier) {
+		return nil, errNilValidatorStatisticsQuerier
+	}
+	if check.IfNil(args.StatusHandler) {
+		return nil, errNilStatusHandler
+	}
+	if check.IfNil(args.BlsKeysFetcher) {
+		return nil, errNilBLSKeysFetcher
+	}
+
+	return &blsKeysExecutor{
+		notifiers:                  args.Notifiers,
+		ratingsChecker:             args.RatingsChecker,
+		validatorStatisticsQuerier: args.ValidatorStatisticsQuerier,
+		statusHandler:              args.StatusHandler,
+		name:                       args.Name,
+		explorerURL:                args.ExplorerURL,
+		blsKeysFetcher:             args.BlsKeysFetcher,
+	}, nil
+}
+
+// Execute executes one checking cycle
+func (executor *blsKeysExecutor) Execute(ctx context.Context) error {
+	log.Debug("executing query-check-notify cycle", "executor", executor.name)
+	statistics, err := executor.validatorStatisticsQuerier.Query(ctx)
+	if err != nil {
+		log.Debug("blsKeysExecutor.Execute", "executor", executor.name, "error calling query", err.Error())
+		executor.statusHandler.ErrorEncountered(err)
+
+		return err
+	}
+
+	extraBLSKeys, err := executor.blsKeysFetcher.GetAllBLSKeys(ctx, executor.name)
+	if err != nil {
+		log.Debug("blsKeysExecutor.Execute", "executor", executor.name, "error calling GetAllBLSKeys", err.Error())
+		executor.statusHandler.ErrorEncountered(err)
+
+		return err
+	}
+
+	problematicKeys, err := executor.ratingsChecker.Check(statistics, extraBLSKeys)
+	if err != nil {
+		log.Debug("blsKeysExecutor.Execute", "executor", executor.name, "error calling check", err.Error())
+		executor.statusHandler.ErrorEncountered(err)
+
+		return err
+	}
+
+	if len(problematicKeys) == 0 {
+		log.Debug("all keys are performing normally", "executor", executor.name)
+
+		return nil
+	}
+
+	messages := executor.createMessages(problematicKeys)
+	executor.notify(messages)
+
+	return nil
+}
+
+func (executor *blsKeysExecutor) createMessages(problematicKeys []core.CheckResponse) []core.OutputMessage {
+	result := make([]core.OutputMessage, 0, len(problematicKeys))
+	for _, key := range problematicKeys {
+		message := core.OutputMessage{
+			Type:               core.ErrorMessageOutputType,
+			IdentifierType:     identifierType,
+			Identifier:         key.HexBLSKey,
+			ShortIdentifier:    shortIdentifier(key.HexBLSKey),
+			IdentifierURL:      executor.createIdentifierURL(key.HexBLSKey),
+			ExecutorName:       executor.name,
+			ProblemEncountered: key.Status,
+		}
+
+		result = append(result, message)
+	}
+
+	return result
+}
+
+func (executor *blsKeysExecutor) notify(messages []core.OutputMessage) {
+	log.Debug("blsKeysExecutor.notify", "executor", executor.name, "num messages", len(messages))
+
+	executor.statusHandler.ProblematicBLSKeysFound(messages)
+	for _, notifier := range executor.notifiers {
+		notifier.OutputMessages(messages...)
+	}
+}
+
+func shortIdentifier(identifier string) string {
+	if len(identifier) <= 15 {
```

Comment:
maybe you can add a const for this magic number

Reply:
refactored